commit 073a8f491f8707458a4e7ac80a8f8d5743eb57b9
Author: Patrick Brunschwig <patrick@enigmail.net>
Date:   Thu Jan 3 18:00:43 2013 +0100

    ensure CRLF at end of last line (Bug reported by Olav from AppleMail)

diff --git a/package/mimeDecrypt.js b/package/mimeDecrypt.js
index f92b2f1..c9e5e68 100644
--- a/extensions/enigmail/package/mimeDecrypt.js
+++ b/extensions/enigmail/package/mimeDecrypt.js
@@ -249,6 +249,11 @@ PgpMimeDecrypt.prototype = {
     if (gDebugLogLevel > 4)
       DEBUG_LOG("mimeDecrypt.js: done: decrypted data='"+this.decryptedData+"'\n");
 
+    // ensure newline at the end of the stream
+    if (! this.decryptedData.endsWith("\n")) {
+      this.decryptedData +="\r\n";
+      this.dataLength += 2;
+    }
     gConv.setData(this.decryptedData, this.dataLength);
     this.mimeSvc.onDataAvailable(null, null, gConv, 0, this.dataLength);
 
